### ISSUE TITLE
Question - load tree 

### ISSUE THREAD

**Issue by rodgomesc** [NeekSandhu/react-aspen#6]
any docs about how to load tree with a javascript object instead of using system patchs ?

like:

```js
const data = {
  id: 'root',
  name: 'Parent',
  children: [
    {
      id: '1',
      name: 'Child - 1',
    },
    {
      id: '3',
      name: 'Child - 3',
      children: [
        {
          id: '4',
          name: 'Child - 4',
        },
      ],
    },
  ],
};
```

**Comment by NeekSandhu** [comment]
If you don't need renaming and file creation capabilities, you can create a basic "filesystem" adapter. `TreeModel` requires a host with function `getItems` which will be called initially for the root path, and then later as the user expands directories. The path will be formatted depending on what you specify for `pathStyle` i.e `win32` or `unix`, latter recommended.

Then you need to rig your `getItems` callback to translate requested path to actual location of entity in your "object". Probably by traversing recursively and drilling down by each "path segment" until you hit the thing. You can use `splitPath` from `path-fx` library to split a path like `"/root/Child - 3/Child - 4"` to an array of keys like `["root", "Child - 3", "Child - 4"]`.

Then just walk the keys recursively.

___

See also:

https://github.com/NeekSandhu/react-aspen-demo/blob/master/demo/index.tsx#L25-L38

https://github.com/NeekSandhu/react-aspen-demo/blob/master/demo/index.tsx#L40

https://github.com/NeekSandhu/react-aspen-demo/blob/master/demo/index.tsx#L70

___

Basically you should just fork the [`react-aspen-demo`](https://github.com/NeekSandhu/react-aspen-demo/) repo to get a boost

**Comment by NeekSandhu** [comment]
Did this solve your problem?

**Comment by NeekSandhu** [comment]
Closing due to inactivity
